source/fundamentals/object-encodings.txt

```diff
@@ -0,0 +1,98 @@
+===============
+Object Encoding
+===============
+
+.. default-domain:: mongodb
+
+Overview
+--------
+
+You can customize the class that represents BSON documents in a collection
+using an optional second parameter of the
+:java-sync-api:`MongoDatabase.getCollection()
+<com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`
+method.
+
+To use a class in this way, a :java-docs:`Codec
+<apidocs/bson/org/bson/codecs/Codec.html>` for it must be registered in the
+:java-docs:`CodecRegistry
+<apidocs/bson/org/bson/codecs/configuration/CodecRegistry.html>`
+of the ``MongoCollection``.
+
```

Comment:
Could we explain and provide simple snippets of how Codecs, CodecRegistries, and CodecProviders are used similar to the original page? I think it would be useful to users that might not have any context yet but would like to learn.